x/oracle/types/msgs.go

```diff
@@ -168,3 +172,115 @@ func (msg MsgDelegateFeedConsent) ValidateBasic() error {
+func (msg MsgGovUpdateParams) String() string {
+	out, _ := yaml.Marshal(msg)
+	return string(out)
+}
+
+// GetSignBytes implements Msg
+func (msg MsgGovUpdateParams) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(&msg)
+	return sdk.MustSortJSON(bz)
+}
+
+// GetSigners implements Msg
+func (msg MsgGovUpdateParams) GetSigners() []sdk.AccAddress {
+	return checkers.Signers(msg.Authority)
+}
+
+// ValidateBasic implements Msg
+func (msg MsgGovUpdateParams) ValidateBasic() error {
+	if err := checkers.ValidateProposal(msg.Title, msg.Description, msg.Authority); err != nil {
+		return err
+	}
+
+	for _, key := range msg.Keys {
+		switch key {
+		case string(KeyVotePeriod):
+			if msg.Changes.VotePeriod == 0 {
+				return fmt.Errorf("oracle parameter VotePeriod must be > 0")
+			}
+
+		case string(KeyVoteThreshold):
+			if msg.Changes.VoteThreshold.LTE(sdk.NewDecWithPrec(33, 2)) {
+				return fmt.Errorf("oracle parameter VoteThreshold must be greater than 33 percent")
+			}
+
+		case string(KeyRewardBands):
+			for _, v := range msg.Changes.RewardBands {
+				if v.RewardBand.GT(sdk.OneDec()) || v.RewardBand.IsNegative() {
+					return fmt.Errorf("oracle parameter RewardBand must be between [0, 1]")
+				}
+				if len(v.SymbolDenom) == 0 {
+					return fmt.Errorf("oracle parameter RewardBand must have SymbolDenom")
+				}
+			}
+
+		case string(KeyRewardDistributionWindow):
+			if msg.Changes.RewardDistributionWindow == 0 {
+				return fmt.Errorf("oracle parameter RewardDistributionWindow must be > 0")
+			}
+
+		case string(KeyAcceptList):
+			for _, denom := range msg.Changes.AcceptList {
+				if len(denom.BaseDenom) == 0 {
+					return fmt.Errorf("oracle parameter AcceptList Denom must have BaseDenom")
+				}
+				if len(denom.SymbolDenom) == 0 {
+					return fmt.Errorf("oracle parameter AcceptList Denom must have SymbolDenom")
+				}
+			}
+
+		case string(KeyMandatoryList):
+			for _, denom := range msg.Changes.MandatoryList {
+				if len(denom.BaseDenom) == 0 {
+					return fmt.Errorf("oracle parameter MandatoryList Denom must have BaseDenom")
+				}
+				if len(denom.SymbolDenom) == 0 {
+					return fmt.Errorf("oracle parameter MandatoryList Denom must have SymbolDenom")
+				}
+			}
+
+		case string(KeySlashFraction):
+			if msg.Changes.SlashFraction.GT(sdk.OneDec()) || msg.Changes.SlashFraction.IsNegative() {
+				return fmt.Errorf("oracle parameter SlashFraction must be between [0, 1]")
+			}
+
+		case string(KeySlashWindow):
+			if msg.Changes.SlashWindow == 0 {
+				return fmt.Errorf("oracle parameter SlashWindow must be > 0")
+			}
+
+		case string(KeyMinValidPerWindow):
+			if msg.Changes.MinValidPerWindow.GT(sdk.OneDec()) || msg.Changes.MinValidPerWindow.IsNegative() {
+				return fmt.Errorf("oracle parameter MinValidPerWindow must be between [0, 1]")
+			}
+
+		case string(KeyHistoricStampPeriod):
+		case string(KeyMedianStampPeriod):
+		case string(KeyMaximumPriceStamps):
+		case string(KeyMaximumMedianStamps):
```

Comment:
We also need validation for these right?

Reply:
The max stamp params have never had any type of validation thresholds and I don't think they need it. Price and Median stamp are validated in `msg_server.go` in `GovUpdateParams` since they are validated based on each other's values (this `ValidateBasic` method for the actual msg object here doesn't have keeper access so it  wouldn't be able to reference the what the other value is set to). These cases are left in to enforce that these are valid param keys (if a non-existing key is passed the default case handles that).

Reply:
Do we not need validation that they're non-negative and non-zero?

Reply:
I noticed we have validate methods for these params in the types package so I changed this method a bit to use those